drydock/patches/kustomization

```diff
@@ -9,6 +9,11 @@ patches:
 - path: plugins/drydock/k8s/lifecycle/lms.yml
 - path: plugins/drydock/k8s/lifecycle/cms.yml
 {% endif -%}
+{% if DRYDOCK_REGISTRY_CREDENTIALS -%}
+- target:
+    kind: Deployment
```

Comment:
You should probably take into account other kind of workloads: jobs, daemonsets and statefulsets. 

A job may pass if its image is also used in a deployment and the image get cached in the node, but if the image is only used for jobs then the pull secret won't be configured.